database/migrations/Souk/2025_10_26_000006_add_missing_trigger_types_to_loyalty_offers.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Schema;
+
+return new class () extends Migration {
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        // Drop the old enum constraint and create a new one with all trigger types
+        Schema::connection('commerce')->table('loyalty_offers', function (Blueprint $table) {
+            // For MySQL, we need to modify the column type
+            // This changes the ENUM values to include the missing trigger types
+            DB::connection('commerce')->statement(
+                "ALTER TABLE loyalty_offers MODIFY trigger_type ENUM(
```

Comment:
It seems that this could be customized by companies or apps, doesn't it? In the future it could be its own table to have more flexibility